src/bun.js/bindings/v8/V8Object.cpp

```diff
@@ -62,10 +62,10 @@ Maybe<bool> Object::Set(Local<Context> context, Local<Value> key, Local<Value> v
 
 void Object::SetInternalField(int index, Local<Data> data)
 {
-    auto fields = getInternalFieldsContainer(this);
-    if (fields && index >= 0 && index < fields->size()) {
-        fields->at(index) = InternalFieldObject::InternalField(data->localToJSValue(Isolate::GetCurrent()->globalInternals()));
-    }
+    auto* fields = getInternalFieldsContainer(this);
+    RELEASE_ASSERT(fields, "object has no internal fields");
```

Comment:
Can these be `ASSERT`? This will be expensive in a release build

Reply:
This can be triggered by modules that misuse the V8 API. I think it's better in those cases to have a more descriptive error message rather than segfaulting. Also, segfault will show the "this is an error in Bun not your code" message which isn't really true.